Remove ENABLE_MOUSE_FORCE_EVENTS directive

annevk opened the pull request:
#### 5256dc75bf2e7720c233fded2a89fd99eab96825
<pre>
Remove ENABLE_MOUSE_FORCE_EVENTS directive
<a href="https://bugs.webkit.org/show_bug.cgi?id=268079">https://bugs.webkit.org/show_bug.cgi?id=268079</a>

Reviewed by Ryosuke Niwa and Darin Adler.

This has the benefit of allowing us to add test coverage which reveals
we are inconsistent between the body and frameset elements. This PR
does not address that however. It&apos;s a pure cleanup.

* LayoutTests/fast/dom/event-handler-attributes-expected.txt:
* LayoutTests/fast/dom/event-handler-attributes.html:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::dispatchMouseForceWillBegin):
* Source/WebCore/dom/GlobalEventHandlers.idl:
* Source/WebCore/dom/MouseEvent.idl:
* Source/WebCore/html/HTMLBodyElement.idl:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleMouseForceEvent):

Canonical link: <a href="https://commits.webkit.org/273603@main">https://commits.webkit.org/273603@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c97570442af1b07a59c1c6130019c89c4bd5aab5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35826 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14769 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37978 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38550 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32230 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/37045 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17340 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11794 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30979 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36381 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12610 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31851 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10963 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/11077 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32024 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39799 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/30303 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32678 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32346 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36911 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/35678 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11143 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9041 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35002 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12892 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/31688 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/42384 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8196 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11648 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/8786 "Found 39 new JSC stress test failures: microbenchmarks/array-from-derived-object-func.js.bytecode-cache, microbenchmarks/array-from-derived-object-func.js.default, microbenchmarks/array-from-derived-object-func.js.dfg-eager, microbenchmarks/array-from-derived-object-func.js.dfg-eager-no-cjit-validate, microbenchmarks/array-from-derived-object-func.js.eager-jettison-no-cjit, microbenchmarks/array-from-derived-object-func.js.no-cjit-collect-continuously, microbenchmarks/array-from-derived-object-func.js.no-cjit-validate-phases, microbenchmarks/array-from-derived-object-func.js.no-llint, microbenchmarks/array-from-object-func.js.bytecode-cache, microbenchmarks/array-from-object-func.js.default ... (failure)") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11973 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (commented):
I see existing uses for `ENABLE(MOUSE_FORCE_EVENTS)` in some IDL files, WebCore::EventHandler, and WebCore::Element though! Unless I'm misunderstanding?

```
Source/WebCore/page/EventHandler.cpp
2334:#if ENABLE(MOUSE_FORCE_EVENTS)
2373:#endif // #if ENABLE(MOUSE_FORCE_EVENTS)

Source/WTF/wtf/PlatformEnable.h
386:#if !defined(ENABLE_MOUSE_FORCE_EVENTS)
387:#define ENABLE_MOUSE_FORCE_EVENTS 1

Source/WebCore/dom/Element.cpp
3724:#if ENABLE(MOUSE_FORCE_EVENTS)

Source/WebCore/dom/GlobalEventHandlers.idl
115:    [NotEnumerable, Conditional=MOUSE_FORCE_EVENTS] attribute EventHandler onwebkitmouseforcechanged;
116:    [NotEnumerable, Conditional=MOUSE_FORCE_EVENTS] attribute EventHandler onwebkitmouseforcedown;
117:    [NotEnumerable, Conditional=MOUSE_FORCE_EVENTS] attribute EventHandler onwebkitmouseforcewillbegin;
118:    [NotEnumerable, Conditional=MOUSE_FORCE_EVENTS] attribute EventHandler onwebkitmouseforceup;

Source/WebCore/dom/MouseEvent.idl
26:    [Conditional=MOUSE_FORCE_EVENTS] const double WEBKIT_FORCE_AT_MOUSE_DOWN = 1;
27:    [Conditional=MOUSE_FORCE_EVENTS] const double WEBKIT_FORCE_AT_FORCE_MOUSE_DOWN = 2;
45:    [Conditional=MOUSE_FORCE_EVENTS, ImplementedAs=force]readonly attribute double webkitForce;

Source/WebCore/html/HTMLBodyElement.idl
46:    [NotEnumerable, WindowEventHandler, Conditional=MOUSE_FORCE_EVENTS] attribute EventHandler onwebkitmouseforcechanged;
47:    [NotEnumerable, WindowEventHandler, Conditional=MOUSE_FORCE_EVENTS] attribute EventHandler onwebkitmouseforcedown;
48:    [NotEnumerable, WindowEventHandler, Conditional=MOUSE_FORCE_EVENTS] attribute EventHandler onwebkitmouseforcewillbegin;
49:    [NotEnumerable, WindowEventHandler, Conditional=MOUSE_FORCE_EVENTS] attribute EventHandler onwebkitmouseforceup;
```

aprotyas reviewed (commented):
Looks good to me!

annevk committed:
Remove ENABLE_MOUSE_FORCE_EVENTS directive
https://bugs.webkit.org/show_bug.cgi?id=268079

Reviewed by Ryosuke Niwa and Darin Adler.

This has the benefit of allowing us to add test coverage which reveals
we are inconsistent between the body and frameset elements. This PR
does not address that however. It's a pure cleanup.

* LayoutTests/fast/dom/event-handler-attributes-expected.txt:
* LayoutTests/fast/dom/event-handler-attributes.html:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::dispatchMouseForceWillBegin):
* Source/WebCore/dom/GlobalEventHandlers.idl:
* Source/WebCore/dom/MouseEvent.idl:
* Source/WebCore/html/HTMLBodyElement.idl:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleMouseForceEvent):

Canonical link: https://commits.webkit.org/273603@main
